src/tools/contextTool.ts

```diff
@@ -0,0 +1,269 @@
+ * 
+ * This function:
+ * 1. Validates the feature slug and agent name
+ * 2. Resolves the workspace path containing the feature
+ * 3. Loads workspace-specific custom instructions from .paw/instructions/
+ * 4. Loads user-level custom instructions from ~/.paw/instructions/
+ * 5. Loads raw WorkflowContext.md content from .paw/work/<feature-slug>/
+ * 
+ * Missing files are handled gracefully and returned with exists=false.
+ * 
+ * @param params - Context parameters with feature_slug and agent_name
+ * @returns Promise resolving to ContextResult with all loaded content
+ * @throws Error if feature slug is invalid, agent name is empty, or no workspace is open
+ */
+export async function getContext(params: ContextParams): Promise<ContextResult> {
+  const { featureSlug, agentName } = validateParams(params);
+
+  const workspacePath = resolveWorkspacePath(featureSlug);
+  if (!workspacePath) {
+    throw new Error('Unable to determine workspace path: no workspace folder is currently open.');
+  }
+
+  const workspaceInstructionsDir = path.join(workspacePath, '.paw', 'instructions');
+  const userInstructionsDir = path.join(os.homedir(), '.paw', 'instructions');
+  const workflowContextPath = path.join(workspacePath, '.paw', 'work', featureSlug, 'WorkflowContext.md');
+
+  const workspaceInstructions = loadCustomInstructions(workspaceInstructionsDir, agentName);
+  const userInstructions = loadCustomInstructions(userInstructionsDir, agentName);
+  const workflowContext = loadWorkflowContext(workflowContextPath);
+
+  return {
+    workspace_instructions: workspaceInstructions,
+    user_instructions: userInstructions,
+    workflow_context: workflowContext,
+  };
+}
+
+/**
+ * Formats an instruction section with title, content, and optional error message.
+ * 
+ * @param title - Section title (e.g., 'Workspace Custom Instructions')
+ * @param status - Instruction status containing content and optional error
+ * @returns Formatted Markdown section
+ */
+function formatInstructionSection(title: string, status: InstructionStatus): string {
+  const parts: string[] = [`## ${title}`];
+
+  if (status.content) {
+    parts.push(status.content);
+  }
+
+  if (status.error) {
+    parts.push(`Warning: ${status.error}`);
+  }
+
+  return parts.join('\n\n');
+}
+
+/**
+ * Formats a complete context result as a natural language Markdown response
+ * suitable for agent consumption.
+ * 
+ * The formatted response includes:
+ * - Precedence rules for custom instructions
+ * - Workspace custom instructions (if present)
+ * - User custom instructions (if present)
+ * - Raw WorkflowContext.md content in a code fence (if present)
+ * - Message if no content was found
+ * 
+ * Empty sections are omitted from the response.
+ * 
+ * @param result - Context result to format
+ * @returns Formatted Markdown text ready for agent consumption
+ */
+export function formatContextResponse(result: ContextResult): string {
+  const sections: string[] = [
```

Comment:
Do not give instructions. The agent prompts will contain instructions on how to interpret the information.